source/javascripts/services/request-service.ts

```diff
@@ -0,0 +1,158 @@
+		if (abortedPromise) {
+			abortedPromise.then(() => {
+				controller.abort();
+			});
+		}
+
+		let requestURL = "";
+
+		switch (this.mode) {
+			case RequestServiceMode.Website:
+				requestURL = StringService.stringReplacedWithParameters(window["routes"].website.yml_get, {
+					app_slug: this.appSlug // eslint-disable-line @typescript-eslint/camelcase
+				});
+
+				break;
+			case RequestServiceMode.Cli:
+				requestURL = window["routes"].local_server.yml_get;
+
+				break;
+		}
+
+		let response: Response;
+		try {
+			response = await fetch(requestURL, {
+				signal: controller.signal
+			});
+		} catch (error) {
+			this.logErrorWithLevel(error, StatusType.warn);
+			throw this.requestAbortedError(window["strings"].request_service.load_app_config.error_prefix);
+		}
+
+		if (!response.ok) {
+			let responseBody: { bitrise_yml?: string; error: string };
+			try {
+				responseBody = await response.json();
+			} catch (error) {
+				this.logErrorWithLevel(error, StatusType.error);
+				throw Error(window["strings"].request_service.load_app_config.default_error);
+			}
+
+			if (responseBody.bitrise_yml) {
+				const error = this.prefixedError(
+					this.errorFromResponseBody(
+						responseBody,
+						window["strings"].request_service.load_app_config.invalid_bitrise_yml_error
+					).message,
+					window["strings"].request_service.load_app_config.error_prefix
+				);
+				this.logErrorWithLevel(error, StatusType.warn);
+
+				throw {
+					bitriseYml: responseBody.bitrise_yml,
+					error: error
+				};
+			}
+
+			const error = this.errorFromResponseBody(
+				responseBody,
+				window["strings"].request_service.load_app_config.default_error
+			);
+			this.logErrorWithLevel(error, response.status < 500 ? StatusType.warn : StatusType.error);
+			throw error;
+		}
+
+		try {
+			return await response.text();
+		} catch (error) {
+			this.logErrorWithLevel(error, StatusType.error);
+			throw new Error(window["strings"].request_service.load_app_config.default_error);
+		}
+	}
+
+	private modeFromEnvVars(): RequestServiceMode {
+		if (process.env["MODE"] === "WEBSITE") {
+			return RequestServiceMode.Website;
+		}
+
+		return RequestServiceMode.Cli;
+	}
+
+	private prefixedError(message: string, messagePrefix: string): Error {
+		return new Error(messagePrefix + message);
+	}
+
+	private requestAbortedError(messagePrefix: string): Error {
+		return this.prefixedError(window["strings"].request_service.request.aborted, messagePrefix);
+	}
+
+	private errorFromResponseBody(
+		responseBody: { error?: string; error_msg?: string },
+		defaultMessage: string = window["strings"].request_service.response.default_error
+	): Error {
+		return new Error(responseBody.error || responseBody.error_msg || defaultMessage);
+	}
+
+	private logErrorWithLevel(error: Error, level: StatusType): void {
+		if (!this.logger) {
+			return;
+		}
+		const logger = (this.logger as unknown) as Logger;
+
+		switch (level) {
```

Comment:
I think we could be simplified, only "error" differs from the others, so
`level===StatusType.error ? logger.error(error) : logger[level](error.message)`

Reply:
Though it would be simpler, I think it's dangerous: the fact that the `StatusType` enum has the same keys as the `Logger` interface's methods is coincidental. If we were to rename methods of the `Logger` interface - which we would assume to be a safe change - it would break this functionality.